lib/pbench/server/api/resources/compare_dataset.py

```diff
@@ -0,0 +1,145 @@
+from http import HTTPStatus
+from urllib.request import Request
+
+from flask import current_app, jsonify
+from flask.wrappers import Response
+from pquisby.lib.post_processing import BenchmarkName, InputType, QuisbyProcessing
+
+from pbench.server import OperationCode, PbenchServerConfig
+from pbench.server.api.resources import (
+    APIAbort,
+    ApiAuthorizationType,
+    ApiBase,
+    ApiContext,
+    APIInternalError,
+    ApiMethod,
+    ApiParams,
+    ApiSchema,
+    convert_dataset,
+    Parameter,
+    ParamType,
+    Schema,
+    UnauthorizedAccess,
+)
+import pbench.server.auth.auth as Auth
+from pbench.server.cache_manager import (
+    CacheManager,
+    TarballNotFound,
+    TarballUnpackError,
+)
+from pbench.server.database import Dataset
+from pbench.server.database.models.users import User
+
+
+class CompareDataset(ApiBase):
+    """
+    This class implements the Server API used to retrieve comparison data for visualization.
+    """
+
+    def __init__(self, config: PbenchServerConfig):
+        super().__init__(
+            config,
+            ApiSchema(
+                ApiMethod.GET,
+                OperationCode.READ,
+                query_schema=Schema(
+                    Parameter(
+                        "datasets", ParamType.LIST, string_list=",", required=True
+                    ),
+                ),
+                authorization=ApiAuthorizationType.NONE,
```

Comment:
This is interesting. In theory maybe `ApiAuthorizationType.DATASET` could be made to work here by recognizing that it's a list and validating *all* of them; but I'm pretty sure it won't do that.

Without that, however, you will have to manually authenticate against each dataset in the list to ensure that the user has READ access. Something like

```python
        self._check_authorization(
            ApiAuthorization(
                ApiAuthorizationType.USER_ACCESS,
                OperationCode.READ,
                str(dataset.owner_id),
                dataset.access,
            )
        )
```

I think the code as written now will allow comparing anyone's private datasets.